README.md

```diff
@@ -14,3 +14,21 @@ part of the array each recursive call considers.
 Analyse the time complexity of your implementation and give a $\Theta$ bound for
 its worst-case runtime. Add your answer, including your reasoning, to this
 markdown file.
+
+# Solution
+
+First let's look at the outer while loop (line 7). Here we iterate until width reaches $n$ (length of array). Width grows by a factor of 2 times every iteration, so it becomes the following series: 1,2,4,8,16,32...,$n$. So to find the total number of iterations we set up the following equation using the geometric sequence: $2^k \ge n$. Solving for $k$ we get $k = \log_2(n)$ which then asymptotically simplifies to $\log(n)$. 
+
+The inner for loop (line 9) runs through the entire array so it simply runs $n$ times. So, combining these we get 
+
+$$\Theta (n \cdot \log(n))$$
+
+# Disclaimer
+
+ I used [this](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/min) so I didn't need to make my own function.
+
+ I also used [this](https://en.wikipedia.org/wiki/Merge_sort) to verify my answer.
+
+ I was confused about what in-place meant, so I used [this](https://www.geeksforgeeks.org/in-place-merge-sort/) for the following quote: "In-place means it does not occupy extra memory for merge operation as in the standard case."
```

Comment:
In place means that the overall algorithm (not just the merge operation) needs at most a constant amount of extra memory.